File: Articalorias.UI/src/types/foodTemplate.ts
import type { MacroAmounts } from './macros';

export interface FoodTemplateResponse {
  foodTemplateId: number;
  templateName: string;
  portionDescription: string;
  defaultQuantity: number;
  caloriesKcal: number;
  /** Amounts PER 1 PORTION keyed by macro key (absent = not captured). */
  macros: MacroAmounts;
  autoAddToNewDay: boolean;
  isActive: boolean;
}

export interface CreateFoodTemplateRequest {
  templateName: string;
  portionDescription: string;
  defaultQuantity: number;
  caloriesKcal: number;
  /** Amounts PER 1 PORTION keyed by macro key. */
  macros: MacroAmounts;
  autoAddToNewDay: boolean;
}

export interface UpdateFoodTemplateRequest {
  templateName: string;
  portionDescription: string;
  defaultQuantity: number;
  caloriesKcal: number;
  /** Amounts PER 1 PORTION keyed by macro key. */
  macros: MacroAmounts;
  autoAddToNewDay: boolean;
}

File: Articalorias/Models/Entities/FoodTemplate.cs
using Articalorias.Services.Macros;

namespace Articalorias.Models.Entities;

public class FoodTemplate
{
    public long FoodTemplateId { get; set; }
    public long UserId { get; set; }
    public string TemplateName { get; set; } = string.Empty;
    public string PortionDescription { get; set; } = string.Empty;
    public decimal DefaultQuantity { get; set; } = 1m;
    public decimal CaloriesKcal { get; set; }

    /// <summary>
    /// Amounts PER 1 PORTION keyed by catalog macro key (see
    /// <see cref="Services.FoodTemplateMath"/> for the scaling rule when a
    /// template becomes an entry). Absent key = not captured when the
    /// template was saved.
    /// </summary>
    public MacroAmounts Macros { get; set; } = MacroAmounts.Empty;

    public bool AutoAddToNewDay { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAtUtc { get; set; }
    public DateTime UpdatedAtUtc { get; set; }
    public byte[] RowVersion { get; set; } = [];

    // Navigation
    public User User { get; set; } = null!;
    public ICollection<FoodEntry> FoodEntries { get; set; } = [];
}
